Code/Editor/CryEdit.cpp

```diff
@@ -3959,32 +3960,64 @@ void CCryEditApp::OpenLUAEditor(const char* files)
         {
             for (const auto& resolvedPath : resolvedPaths)
             {
-                args.append(AZStd::string::format(" -files %s", resolvedPath.c_str()));
+                args.append(AZStd::string::format("\"-files=%s\"", resolvedPath.c_str()));
+                fileName = resolvedPath.c_str();
             }
         }
     }
 
-    AZ::IO::FixedMaxPathString engineRoot = AZ::Utils::GetEnginePath();
-    AZ_Assert(!engineRoot.empty(), "Unable to query Engine Path");
+    const char* engineRoot = nullptr;
+    AzFramework::ApplicationRequests::Bus::BroadcastResult(engineRoot, &AzFramework::ApplicationRequests::GetEngineRoot);
+    AZ_Assert(engineRoot != nullptr, "Unable to communicate to AzFramework::ApplicationRequests::Bus");
+    AZStd::string externalProcess = gSettings.textEditorForScript.toUtf8().data();
+    AZStd::string process;
+    AZStd::string processArgs;
 
+    AZStd::string projectPath;
+    if (auto settingsRegistry = AZ::SettingsRegistry::Get(); settingsRegistry != nullptr)
+    {
+        settingsRegistry->Get(projectPath, AZ::SettingsRegistryMergeUtils::FilePathKey_ProjectPath);        
+    }
+    
+    // only launch IDEA for Lua Script
+    if (externalProcess.ends_with("idea64.exe") || externalProcess.ends_with("idea.exe") || externalProcess.ends_with("idea.sh"))
+    {
+        process = "\"" + externalProcess + "\"";
+        const auto projectName = AZ::Utils::GetProjectName();
+        AZStd::string projectRoot = AZStd::string::format("%s/%s", engineRoot, projectName.c_str());
+        if (fileName.empty())
+        {
+            processArgs = projectRoot;
+        }
```

Comment:
No need to initialize ```projectRoot``` for cases when its not used. Also suggest to use ```BasicPath<AZStd::string>``` here as well.
```suggestion
        if (fileName.empty())
        {
            AZ::IO::BasicPath<AZStd::string> projectRootPath {engineRoot};
            projectRootPath /= AZ::Utils::GetProjectName();
            processArgs = projectRootPath.String();
        }
```